iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/node/WALNode.java

```diff
@@ -215,38 +218,64 @@ public void deleteOutdatedFiles() {
   }
 
   private class DeleteOutdatedFileTask implements Runnable {
-    private static final int MAX_RECURSION_TIME = 5;
     // .wal files whose version ids are less than first valid version id should be deleted
     private long firstValidVersionId;
     // the effective information ratio
-    private double effectiveInfoRatio;
-    // recursion time of calling deletion
-    private int recursionTime = 0;
```

Comment:
It's unsuitable to remove the retry procedure, because outdated wal files may be kept by several memtables and the retry procedure can snapshot/flush more outdated memtables in one task.